src/components/component.ts

```diff
@@ -65,38 +122,23 @@ export class Component {
    * @param {d3.Selection} selection.
    * @returns {Component} The calling Component.
    */
-  public anchor(selection: SimpleSelection<void>) {
+  public anchor(selection: d3.Selection<HTMLElement, any, any, any>) {
     selection = coerceExternalD3(selection);
     if (this._destroyed) {
       throw new Error("Can't reuse destroy()-ed Components!");
     }
 
-    this._isTopLevelComponent = (<Node> selection.node()).nodeName.toLowerCase() === "svg";
-
-    if (this._isTopLevelComponent) {
-      // svg node gets the "plottable" CSS class
-      this._rootSVG = selection;
-      this._rootSVG.classed("plottable", true);
-      // visible overflow for firefox https://stackoverflow.com/questions/5926986/why-does-firefox-appear-to-truncate-embedded-svgs
-      this._rootSVG.style("overflow", "visible");
-
-      // HACKHACK: Safari fails to register events on the <svg> itself
-      const safariBacking = this._rootSVG.select(`.${Component._SAFARI_EVENT_BACKING_CLASS}`);
-      if (safariBacking.empty()) {
-        this._rootSVG.append("rect").classed(Component._SAFARI_EVENT_BACKING_CLASS, true).attrs({
-          x: 0,
-          y: 0,
-          width: "100%",
-          height: "100%",
-        }).style("opacity", 0);
-      }
+    if (this.parent() == null) {
```

Comment:
Yeah there's three equivalences:

`this.parent() == null` <-> `this._rootElement != null` <-> this is the top-level component. I'll add a method `isTopLevel()`?

Aside: we should do a pass over the terminology. I've seen "top level component", "top level element", "root element", "root component", as well as `component._element` and "root DOM node"